src/component/feature/FeatureStrategy/FeatureStrategyConstraints2/useConstraintsValidation.ts

```diff
@@ -0,0 +1,30 @@
+import useFeatureApi from 'hooks/api/actions/useFeatureApi/useFeatureApi';
+import { useEffect, useState } from 'react';
+import { IConstraint } from 'interfaces/strategy';
+
+export const useConstraintsValidation = (
+    projectId: string,
+    featureId: string,
+    constraints?: IConstraint[]
+): boolean => {
+    const [valid, setValid] = useState(constraints?.length === 0);
```

Comment:
Are the constraints automatically valid if the list is greater than one? I guess this is for the editing mode. Would it be simpler to return an async function that accepts a list of constraints and essentially does what is done in the useEffect?

Reply:
The idea is that an empty list of constraints is valid, while a non-empty list has to be checked. Added a comment.

The useState/useEffect handles running validations whenever constraints change, storing the latest result.